oauth.go

```diff
@@ -40,3 +45,33 @@ func (app App) GetAccessToken(shopName string, code string) (string, error) {
 	err = client.Do(req, token)
 	return token.Token, err
 }
+
+// Verify a message against a message HMAC
+func (app App) VerifyMessage(message, messageMAC string) bool {
```

Comment:
`app` is an [`App`](https://github.com/Receiptful/go-shopify/blob/master/goshopify.go#L17-L22). Two things to note about that:

1. `App` is available because the package is the same across all files: `goshopify`. Everything that is defined in the same package is available globally.
2. When defining the function with `(app App)` in the beginning, we are defining the function on the `App` type. In a sense, `app` then becomes like `this` or `self` from other languages. The same function signature in C# would have been something like:

  ```csharp
  class App {
    public bool VerifyMessage(string message, string messageMAC) {
      // I can use this.apiSecret in here
      ...
    }
  }
  ```
  I hope no golang veterans are ever reading this, because I'm probably a bit off on the concepts :-)

Regarding argument types: If multiple arguments have the same type, you can leave out the type declaration until the last in the list of arguments. For example, if I added an `int` argument between `message` and `messageMAC`, it would be an error. I.e. if I do this:
`func (app App) VerifyMessage(message, hej int, messageMAC string) bool {}`
Then the function now thinks `message` is an `int` and the tests give the following error:

```
sudo docker-compose run --rm test
# github.com/receiptful/go-shopify
./oauth.go:52: cannot convert message (type int) to type []byte
./oauth.go:70: cannot use message (type string) as type int in argument to app.VerifyMessage
```